sdk/storage/Azure.Storage.Common/src/Shared/PooledMemoryStream.cs

```diff
@@ -315,16 +326,44 @@ public override void SetLength(long value)
 
         public override void Write(byte[] buffer, int offset, int count)
         {
-            throw new NotSupportedException();
+            do
+            {
+                BufferPartition currentBuffer = GetLatestBufferWithAvailableSpaceOrDefault();
+
+                if (currentBuffer == default)
+                {
+                    byte[] newBytes = ArrayPool.Rent(ArrayPoolRentalSize);
+                    currentBuffer = new BufferPartition
+                    {
+                        Buffer = newBytes,
+                        DataLength = 0
+                    };
+                    BufferSet.Add(currentBuffer);
+                }
+
+                int copied = Math.Min(currentBuffer.Buffer.Length - currentBuffer.DataLength, count);
+                Array.Copy(buffer, offset, currentBuffer.Buffer, currentBuffer.DataLength, copied);
+                currentBuffer.DataLength += copied;
+                count -= copied;
+                offset += copied;
+                Position += copied;
+            }
+            while (count > 0);
```

Comment:
Nit: if someone writes logic that calls this method with count = 0 as part of their own loop, I'd rather have caught that at the start and avoided all these operations in exchange for one extra check at the start of each write.

Reply:
I'll flip it to a normal while loop.